.github/workflows/ci.yml

```
@@ -22,16 +22,16 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - uses: pdm-project/setup-pdm@v4
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v3
         with:
-          python-version: ${{ matrix.python-version }}
-          cache: true
-          allow-python-prereleases: true
+          enable-cache: true
+
+      - name: Set up Python ${{ matrix.python-version }}
+        run: uv python install ${{ matrix.python-version }}
 
       - name: install
-        run: |
-          pdm venv create --with-pip --force $PYTHON
-          pdm install -G linting -G email
+        run: uv sync --python ${{ matrix.python-version }} --group linting --extra email
```

```suggestion
        run: uv sync --python ${{ matrix.python-version }} --group linting
```
Don't need to specify email from extra dependency, it's added in `dev` group and dev group automatically installed with uv sync